### ISSUE TITLE
Non-SGD Optimizer for GradDotProd LM Example

### ISSUE THREAD

**Issue by won-bae** [Jiachen-T-Wang/GhostSuite#7]
Hi, 

Thank you for sharing a detailed code for such an inspiring paper; personally enjoyed reading the paper a lot. 
You mentioned that one of limitations of this work is that it is not straightforward to apply efficient ghost dot-product trick for other optimizers such as Adam due to its normalization. But it seems like AdamW was used as a default optimizer for this example (resulting in Figure 2). 

Have you had a chance to investigate the difference in shapley scores computed based on SGD proxy vs. AdamW (inefficient computation)? I wonder how good SGD proxy is compared to using a "right" optimizer given that change in loss can be quite significant even for one training step between SGD and AdamW.

Thank you.

**Comment by Ruby1R1ng** [comment]
> Hi,
> 
> Thank you for sharing a detailed code for such an inspiring paper; personally enjoyed reading the paper a lot. You mentioned that one of limitations of this work is that it is not straightforward to apply efficient ghost dot-product trick for other optimizers such as Adam due to its normalization. But it seems like AdamW was used as a default optimizer for this example (resulting in Figure 2).
> 
> Have you had a chance to investigate the difference in shapley scores computed based on SGD proxy vs. AdamW (inefficient computation)? I wonder how good SGD proxy is compared to using a "right" optimizer given that change in loss can be quite significant even for one training step between SGD and AdamW.
> 
> Thank you.


I've recently read a paper titled OPUS: Towards Efficient and Principled Data Selection in Large Language Model Pre-training in Every Iteration (available at https://arxiv.org/pdf/2602.05400). It focuses on efficient and principled data selection during LLM pre-training using AdamW and Muon.
